tmk_core/common/chibios/eeprom_stm32.c

```diff
@@ -14,185 +14,793 @@
  * Artur F.
  *
  * Modifications for QMK and STM32F303 by Yiancar
+ * Modifications to add flash wear leveling by Ilya Zhuravlev
+ * Modifications to increase flash density by Don Kjer
  */
 
 #include <stdio.h>
-#include <string.h>
+#include <stdbool.h>
 #include "eeprom_stm32.h"
-/*****************************************************************************
- * Allows to use the internal flash to store non volatile data. To initialize
- * the functionality use the EEPROM_Init() function. Be sure that by reprogramming
- * of the controller just affected pages will be deleted. In other case the non
- * volatile data will be lost.
- ******************************************************************************/
+#include "flash_stm32.h"
+#ifdef FLASH_STM32_MOCKED
+#    include "flash_stm32_mock.h"
+#endif
+#ifdef DEBUG_EEPROM
+#    ifdef FLASH_STM32_MOCKED
+#        define dprintf printf
```

Comment:
Im thinking something like #13556 would be nicer here

Reply:
I have removed `DEBUG_EEPROM` in favor of using the `debug_eeprom` flag